File: apps/pwabuilder/Validations/Models/Validation.cs
namespace PWABuilder.Validations.Models
{
    public class Validation
    {
        public string? Member { get; set; }
        public string? Category { get; set; }
        public string? DisplayString { get; set; }
        public string? ErrorString { get; set; }
        public string? InfoString { get; set; }
        public object? DefaultValue { get; set; }
        public Uri? DocsLink { get; set; }
        public bool? QuickFix { get; set; }
        public bool? TestRequired { get; set; }
        public bool? Valid { get; set; }
    }
}

File: apps/pwabuilder/Frontend/src/script/models/validation.ts
// Local copy of the Validation type originally sourced from
// @pwabuilder/manifest-validation. Copied here so the frontend can drop the
// runtime dependency on that package for type-only usage.

export interface Validation {
    infoString?: string;
    displayString?: string;
    category: string;
    member: string;
    defaultValue?: string | boolean | any[];
    docsLink?: string;
    errorString: string;
    quickFix: boolean;
    test?: Function;
    testRequired?: boolean;
    testName?: string;
    valid?: boolean;
}
